src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -57,6 +58,40 @@ public virtual async Task RaisePackagesMissingEventForSolutionAsync(string solut
             PackagesMissingStatusChanged?.Invoke(this, new PackagesMissingStatusEventArgs(missing));
         }
 
+        public virtual async Task<bool> GetMissingAssetsFileStatusAsync()
+        {
+            foreach (var nuGetProject in (await SolutionManager.GetNuGetProjectsAsync()))
+            {
+                if (nuGetProject is BuildIntegratedNuGetProject)
+                {
+                    string assetsFilePath = await (nuGetProject as BuildIntegratedNuGetProject).GetAssetsFilePathAsync();
+                    var fileInfo = new FileInfo(assetsFilePath);
+
+                    if (!fileInfo.Exists)
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public virtual async Task RaiseAssetsFileMissingEventForSolutionAsync(string solutionDirectory, CancellationToken token)
+        {
+            // This method is called by different event handlers.
+            // If the solutionDirectory is null, there's no need to do needless work.
+            // Even though the solution closed even calls the synchronous ClearMissingEventForSolution
+            // there's no guarantee that some weird ordering of events won't make the solutionDirectory null.
+            var missing = false;
```

Comment:
Bool variables should have prefixes like is, shouldBe, are, etc.

```suggestion
            var isMissing = false;
```